src/codem/registration/apply.py

```diff
@@ -164,43 +163,60 @@ def _apply_dsm(self) -> None:
         output_path = os.path.join(self.config["OUTPUT_DIR"], output_name)
 
         # construct pdal pipeline
-        pipe = [{"type": "readers.gdal", "filename": self.aoi_file, "header": "Z"}]
+        pipeline = pdal.Reader.gdal(filename=self.aoi_file, header="Z")
 
         # no nodata is present, filter based on its limits
         if self.aoi_nodata is not None:
-            range_filter_task = {
-                "type": "filters.range",
-                "limits": f"Z![{self.aoi_nodata}:{self.aoi_nodata}]",
-            }
-            pipe.append(range_filter_task)
+            pipeline |= pdal.Filter.range(
+                limits=f"Z![{self.aoi_nodata}:{self.aoi_nodata}]"
+            )
 
         # insert the transform filter to register the AOI
         registration_task = self.get_registration_transformation()
-        if isinstance(registration_task, dict):
-            pipe.append(registration_task)
+        if isinstance(registration_task, pdal.pipeline.Filter):
+            pipeline |= registration_task
         else:
             raise ValueError(
                 f"get_registration_transformation returned {type(registration_task)} "
                 "not a dictionary of strings as needed for the pdal pipeline."
             )
 
-        # pdal writing task
-        writer_task = {
-            "type": "writers.gdal",
+        writer_kwargs = {
             "resolution": self.aoi_resolution,
             "output_type": "idw",
             "filename": output_path,
+            # TODO this might not be right, might be coma separated values!!!!
             "metadata": (
                 f"CODEM_VERSION={__version__},"
-                f"TIFFTAG_IMAGEDESCRIPTION=RegisteredCompliment"
+                "TIFFTAG_IMAGEDESCRIPTION=RegisteredCompliment"
             ),
         }
-        # Add nodata argument only if nodata is actually present
+
+        if self.aoi_area_or_point in ("Area", "Point"):
```

Comment:
Do we have tests for this? It can all be quite finicky and it is easy to flip logic and slid things the wrong way.

Reply:
We do in 102fb4ab12f677e553fd578ac910027056f12e37 (commit might get squashed with others later)